packages/frontend/src/components/dialogs/EditAccountAndPasswordDialog.tsx

```diff
@@ -102,15 +103,29 @@ function EditAccountInner(onClose: DialogProps['onClose']) {
         ]),
       })
 
-      if (confirmed) {
-        update()
+      if (!confirmed) {
+        return
+      }
+    } else if (
+      initial_settings.proxy_enabled !== accountSettings.proxy_enabled &&
+      accountSettings.proxy_enabled === '1'
+    ) {
+      const qr = await BackendRemote.rpc.checkQr(
+        selectedAccountId(),
+        accountSettings.proxy_url
+      )
+      if (qr.kind !== 'proxy') {
+        openDialog(AlertDialog, {
+          message: tx('proxy_invalid'),
+        })
+        return
       }
```

Comment:
`checkQr` can throw` e.g. for the string `socks5:`

```suggestion
      let qr: T.Qr | undefined = undefined
      try {
        qr = await BackendRemote.rpc.checkQr(
          selectedAccountId(),
          accountSettings.proxy_url
        )
      } catch (error) {
        const errorStr =
          error instanceof Error ? error.message : JSON.stringify(error)
        openDialog(AlertDialog, {
          message: tx('proxy_invalid') + '\n' + errorStr,
        })
        return
      }
      if (qr.kind !== 'proxy') {
        openDialog(AlertDialog, {
          message: tx('proxy_invalid'),
        })
        return
      }
```

Reply:
Oh, and `import type { T } from '@deltachat/jsonrpc-client'` would need to be added.